src/Select.js

```diff
@@ -392,7 +394,7 @@ const Select = React.createClass({
 		if (this.props.multi) {
 			this.addValue(value);
 			this.setState({
-				inputValue: '',
+				inputValue: this.props.onSelectResetsInput ? '' : this.state.inputValue,
```

Comment:
When you're leaving the value as-is, there is no reason for the setState, so better put it in an `if` around it